### PR TITLE
fix: adjust sync path for mac os

### PR DESCRIPTION
**Pull request by pratikjagrut**
**What issue type does this pull request address?** (keep at least one, remove the others) 
/kind bugfix

**What does this pull request do? Which issues does it resolve?** (use `resolves #<issue_number>` if possible) 
resolves #
#1743

**Does this pull request has user-facing changes?** (e.g. config changes, new/modified commands, new/modified flags) 
No

**Does this pull request add new dependencies?** 
No

### CODE REVIEW

File: pkg/devspace/services/synccontroller/controller.go [hunk 1]
```diff
@@ -226,10 +233,12 @@ func (c *controller) startSync(options *Options, onInitUploadDone chan struct{},
 	)
 
 	localPath := "."
+	if runtime.GOOS == "darwin" {
```
Comment (FabianKramm) [inline pkg/devspace/services/synccontroller/controller.go:236]: Can you put the runtime.GOOS check inside the function?

File: pkg/devspace/services/synccontroller/controller.go [hunk 2]
```diff
@@ -226,10 +233,12 @@ func (c *controller) startSync(options *Options, onInitUploadDone chan struct{},
...
+		localPath, _ = realWorkDir()
```
Comment (FabianKramm) [inline pkg/devspace/services/synccontroller/controller.go:237]: Either return the error, or make sure the local path is "." again. Please also move this logic into the function

**Review by FabianKramm — CHANGES_REQUESTED** [review]
@pratikjagrut thanks a lot for the PR! Please see the comments below

File: pkg/devspace/services/synccontroller/controller.go (hunk 1 above)
Comment (pratikjagrut) [reply pkg/devspace/services/synccontroller/controller.go:236]: done

File: pkg/devspace/services/synccontroller/controller.go (hunk 2 above)
Comment (pratikjagrut) [reply pkg/devspace/services/synccontroller/controller.go:237]: done

**Review by FabianKramm — APPROVED** [review]
@pratikjagrut thanks! LGTM!

**Pull request merged**